tests/test_graph_components/providers/test_nlu_training_data_provider.py

```diff
@@ -0,0 +1,56 @@
+from typing import Text
+
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.graph_components.providers.nlu_training_data_provider import (
+    NLUTrainingDataProvider,
+)
+from rasa.shared.importers.importer import TrainingDataImporter
+
+
+def test_nlu_training_data_provider(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    config_path: Text,
+    nlu_data_path: Text,
+):
+    # create a resource and an importer
+    resource = Resource("xy")
+    importer = TrainingDataImporter.load_from_config(
+        config_path=config_path, training_data_paths=[nlu_data_path]
+    )
+
+    # check the default configuration is as expected
+    config_1 = NLUTrainingDataProvider.get_default_config()
+    assert config_1["language"] == "en"
+    assert config_1["persist"] is False
+
+    # create a provider without training data
+    provider_1 = NLUTrainingDataProvider.create(
+        NLUTrainingDataProvider.get_default_config(),
+        default_model_storage,
+        resource,
+        default_execution_context,
+    )
+    assert isinstance(provider_1, NLUTrainingDataProvider)
+
+    # check the data provided is as expected
+    data_0 = provider_1.provide({}, importer)
```

Comment:
I think if you'd add the "language" to the first parameter (config) here, then both results should be the same

Reply:
... oh, thought this line was failing, but it is the assert below :D strange..

Reply:
... is "en" the default somewhere? 🤔

Reply:
It was. I changed it.